src/odemis/driver/simcam.py

```diff
@@ -289,19 +329,69 @@ def _simulate(self):
         binning = self.binning.value
         res = self.resolution.value
         pxs_pos = self.translation.value
-        shape = self._img.shape
+
+        if self._stage:
+            pos = self._stage.position.value["x"], self._stage.position.value["y"]
+
+            exshift = numpy.array(self._orig_stage) - numpy.array(pos)
+            exshift = tuple(exshift)
+
+            # #convert to pxl
+            pxs = self._imgs[0].metadata[model.MD_PIXEL_SIZE]
+            X = exshift[0]/pxs[0]
+
+            Y = exshift[1] / pxs[1]
+            exs = (X,Y)
+
+            npos=numpy.array(pxs_pos) + numpy.array(exs)
+            pxs_pos=tuple(npos)
+
+            logging.debug('the position is %s',pxs_pos)
+
+        # Pick the image based on the current excitation wl
+
+        try:
+            current_inwl = self.getMetadata()[model.MD_IN_WL]   # (350e-9, 360e-9)
+            logging.debug('current_inwl %s',current_inwl)
+
+            current_inwl_c = util.fluo.get_center(current_inwl) # 355e-9
+
+            closest = float('inf')
+            _img = self._imgs[0]
+            for im in self._imgs:
+                c = util.fluo.get_center(im.metadata[model.MD_IN_WL])
+                if abs(current_inwl_c - c) < closest:
+                    _img = im
+                    closest = abs(current_inwl_c - c)
+
+            # _img = min(self._imgs, key=lambda im: abs(util.fluo.get_center(im.metadata[model.MD_IN_WL] - current_inwl_c)))
```

Comment:
I think you can mix between these 2 writings, and get something more compact and clearer:
```
def dist_wl(img):
  try:
      c = util.fluo.get_center(im.metadata[model.MD_IN_WL])
      return abs(current_inwl_c - c)
  except KeyError:
      return float('inf')

img = min(self._imgs, key=dist_wl)
```